ion-schema/src/test/kotlin/com/amazon/ionschema/reader/IonSchemaReaderV2_0Tests.kt

```diff
@@ -0,0 +1,324 @@
+            ${'$'}ion_schema_2_0
+            type::{
+              name: type1,
+              type: string,
+              codepoint_length: 3,
+            }
+            type::{
+              name: type2,
+              type: symbol,
+              utf8_byte_length: 5
+            } 
+        """
+        )
+
+        val result = reader.readSchema(ionStream)
+
+        assertTrue(result.isOk())
+        assertEquals(
+            SchemaDocument(
+                id = null,
+                ionSchemaVersion = IonSchemaVersion.v2_0,
+                items = listOf(
+                    SchemaDocument.Item.Type(
+                        NamedTypeDefinition(
+                            typeName = "type1",
+                            typeDefinition = TypeDefinition(
+                                constraints = setOf(
+                                    Constraint.Type(TypeArgument.Reference("string")),
+                                    Constraint.CodepointLength(DiscreteIntRange(3))
+                                )
+                            ),
+                        )
+                    ),
+                    SchemaDocument.Item.Type(
+                        NamedTypeDefinition(
+                            typeName = "type2",
+                            typeDefinition = TypeDefinition(
+                                constraints = setOf(
+                                    Constraint.Type(TypeArgument.Reference("symbol")),
+                                    Constraint.Utf8ByteLength(DiscreteIntRange(5))
+                                )
+                            ),
+                        )
+                    )
+                )
+            ),
+            result.unwrap().first
+        )
+        assertFalse(result.unwrap().second.hasNext())
+    }
+
+    @Test
+    fun `readSchema can read a schema with a header`() {
+        val ionStream = ION.loader.load(
+            """
+            ${'$'}ion_schema_2_0
+            schema_header::{
+              imports: [
+                { id: "foo.isl" }
+              ],
+              user_reserved_fields: {
+                type: [foo],  
+              }
+            }
+        """
+        )
+
+        val result = reader.readSchema(ionStream)
+
+        assertTrue(result.isOk())
+        assertEquals(
+            SchemaDocument(
+                id = null,
+                ionSchemaVersion = IonSchemaVersion.v2_0,
+                items = listOf(
+                    SchemaDocument.Item.Header(
+                        imports = listOf(HeaderImport.Wildcard("foo.isl")),
+                        userReservedFields = UserReservedFields(type = setOf("foo"))
+                    ),
+                )
+            ),
+            result.unwrap().first
+        )
+        assertFalse(result.unwrap().second.hasNext())
+    }
+
+    @Test
+    fun `readSchema can read a schema with a footer and returns an iterator with all of the content after the footer`() {
```

Comment:
(Question) Should we be returning the iterator for all the content thats not part of schema? `readSchema` to me doesn't seem like ti should return anything thats not part of the schema. 
I feel we could provide an alternative method maybe `readSchemaDocument`/`iterateSchemaDocument` which can return these non schema contents.

Reply:
I don't have a good answer, so I'm going to punt on this for now and change it back so that `readSchema` will only return the SchemaDocument.